diff --git a/Makefile b/Makefile
index f7330552..c97a88eb 100644
--- a/Makefile
+++ b/Makefile
@@ -405,9 +405,10 @@ KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 		   -fno-strict-aliasing -fno-common \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
-		   -Werror \
-		   -std=gnu89 $(call cc-option,-fno-PIE)
-
+		   -std=gnu89 $(call cc-option,-fno-PIE) \
+		   -D_FORTIFY_SOURCE=1 \
+		   -DNDEBUG \
+		   -pipe
 
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
@@ -754,7 +754,7 @@ KBUILD_CFLAGS += $(call cc-option, -mno-global-merge,)
 KBUILD_CFLAGS += $(call cc-option, -fcatch-undefined-behavior)
 else
 
-KBUILD_CFLAGS   += -mcpu=exynos-m1.cortex-a53
+KBUILD_CFLAGS   += -mcpu=exynos-m1.cortex-a53+crc+simd+crypto
 
 # These warnings generated too much noise in a regular build.
 # Use make W=1 to enable them (see scripts/Makefile.extrawarn)
